src/language-markdown/leading-hardlines.js

```diff
@@ -0,0 +1,93 @@
+import { hardline } from "../document/builders.js";
+import {
+  INLINE_NODE_TYPES,
+  INLINE_NODE_WRAPPER_TYPES,
+  isPrettierIgnore,
+} from "./utils.js";
+
+/**
+ * @typedef {import("../document/builders.js").Doc} Doc
+ */
+
+const SIBLING_NODE_TYPES = new Set([
+  "listItem",
+  "definition",
+  "footnoteDefinition",
+]);
+
+/**
+ * @param {*} path
+ * @param {*} options
+ * @returns {Doc | null}
+ */
+function leadingHardlines(path, options) {
+  if (!shouldPrePrintHardline(path)) {
+    return null;
+  }
+  let length = 1;
+  if (shouldPrePrintTripleHardline(path)) {
+    length = 3;
```

Comment:
I can't find input that cover this path. I left it anyway.
https://app.codecov.io/gh/prettier/prettier/blob/main/src%2Flanguage-markdown%2Fprinter-markdown.js#L663